apps/trading/components/orders-container/orders-container.tsx

```diff
@@ -0,0 +1,125 @@
+import { useDataGridEvents } from '@vegaprotocol/datagrid';
+import { t } from '@vegaprotocol/i18n';
+import { Filter } from '@vegaprotocol/orders';
+import { OrderListManager } from '@vegaprotocol/orders';
+import { Splash } from '@vegaprotocol/ui-toolkit';
+import { useVegaWallet } from '@vegaprotocol/wallet';
+import type { ColumnState } from 'ag-grid-community';
+import {
+  useMarketClickHandler,
+  useMarketLiquidityClickHandler,
+} from '../../lib/hooks/use-market-click-handler';
+import { create } from 'zustand';
+import { persist, subscribeWithSelector } from 'zustand/middleware';
+
+export interface OrderContainerProps {
+  marketId?: string;
+  filter?: Filter;
+}
+
+export const OrdersContainer = ({ marketId, filter }: OrderContainerProps) => {
+  const { pubKey, isReadOnly } = useVegaWallet();
+  const onMarketClick = useMarketClickHandler(true);
+  const onOrderTypeClick = useMarketLiquidityClickHandler(true);
+  const [gridStore, update] = useOrderListStore((store) => {
+    switch (filter) {
+      case Filter.Open: {
+        return [store.open, store.update];
+      }
+      case Filter.Closed: {
+        return [store.closed, store.update];
+      }
+      case Filter.Rejected: {
+        return [store.rejected, store.update];
+      }
+      default: {
+        return [store.all, store.update];
+      }
+    }
+  });
+  const gridStoreCallbacks = useDataGridEvents(gridStore, (colState) => {
+    update(filter, colState);
+  });
+
+  if (!pubKey) {
+    return <Splash>{t('Please connect Vega wallet')}</Splash>;
+  }
+
+  return (
+    <OrderListManager
+      partyId={pubKey}
+      marketId={marketId}
+      filter={filter}
+      onMarketClick={onMarketClick}
+      onOrderTypeClick={onOrderTypeClick}
+      isReadOnly={isReadOnly}
+      gridProps={gridStoreCallbacks}
+    />
+  );
+};
+
+type Store = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  filterModel?: { [key: string]: any };
+  columnState?: ColumnState[];
+};
+
+const useOrderListStore = create<{
```

Comment:
We can make it more generic


```
import { immer } from 'zustand/middleware/immer';

const makeMultiinstanceTableStore = (name: string) =>
  create<{
    gridState: Record<string, Store>;
    update: (gridStore: Store, key: string) => void;
  }>()(
    persist(
      subscribeWithSelector(
        immer((set) => ({
          gridState: {},
          update: (newState: Store, key) => {
            set((state) => {
              state.gridState[key] = newState;
            });
          },
        }))
      ),
      {
        name,
      }
    )
  );

const useOrderListStore = makeMultiinstanceTableStore('vega_order_list_store');
```
and then use it in component
```
const StoreKey: Record<Filter, string> = {
  [Filter.Closed]: 'closed',
  [Filter.Open]: 'open',
  [Filter.Rejected]: 'rejected',
};

export const OrdersContainer = ({ marketId, filter }: OrderContainerProps) => {
  [...]
  const storeKey = filter ? StoreKey[filter] : 'all';
  const gridStore = useOrderListStore((store) => store.gridState[storeKey]);
  const update = useOrderListStore((store) => store.update);
  const gridStoreCallbacks = useDataGridEvents(gridStore, (colState) => {
    update(colState, storeKey);
  })
  [...]
}

```

Reply:
I think that each pane should use a single store for all its state, right now that only includes grid state but in future it might also have state for other things (formatting, small balances, chart options, etc etc).

We can use slices (as per zustand documentation) to organise the state and share logic.